### PR TITLE
New Feature: debug flag transferred by run_scans.py & confAllChambers.py

### PR DESCRIPTION
**Pull request by bdorney**
## Description

The `--debug` flag was not being transferred by `run_scans.py` or `confAllChambers.py` to the relevant tool.  Additionally the `--debug` flag within tools was not applied uniformly.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Motivation and Context

Trying to troubleshoot hardware problems requires appropriate debugging info

## How Has This Been Tested?

http://cmsonline.cern.ch/cms-elog/1013587

Relevant log files and outputs can be found at:
```
/tmp/GE11-X-S-CERN-0001/
```

On the `cosmicstandtif` machine.
## Checklist:

- [X] My code follows the code style of this project.
- [] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: fastLatency.py [hunk 1]
```diff
@@ -17,16 +17,14 @@
...
 (options, args) = parser.parse_args()
-uhal.setLogLevelTo( uhal.LogLevel.WARNING )
 
 if options.MSPL not in range(1,9):
     print "Invalid MSPL specified: %d, must be in range [1,8]"%(options.MSPL)
     exit(1)
 
 if options.debug:
-    uhal.setLogLevelTo( uhal.LogLevel.INFO )
+    uhal.setLogLevelTo( uhal.LogLevel.DEBUG )
```
Comment (jsturdy) [inline fastLatency.py]: there is pretty much no case where when debugging a tool i'm interested in seeing the `uhal::LogLevel::DEBUG` messages, `uhal::LogLevel::INFO` should be sufficient

File: trimChamber.py [hunk 2]
```diff
@@ -20,10 +20,13 @@
 parser.add_option("--vt1", type="int", dest="vt1",
                   help="VThreshold1 DAC value for all VFATs", metavar="vt1", default=100)
 
-
-uhal.setLogLevelTo( uhal.LogLevel.WARNING )
 (options, args) = parser.parse_args()
 
+if options.debug:
+    uhal.setLogLevelTo( uhal.LogLevel.DEBUG )
```
Comment (jsturdy) [inline trimChamber.py]: there is pretty much no case where when debugging a tool i'm interested in seeing the `uhal::LogLevel::DEBUG` messages, `uhal::LogLevel::INFO` should be sufficient

File: ultraLatency.py [hunk 3]
```diff
@@ -64,7 +61,7 @@
     step = options.scanmax - options.scanmin
 
 if options.debug:
-    uhal.setLogLevelTo(uhal.LogLevel.INFO)
+    uhal.setLogLevelTo(uhal.LogLevel.DEBUG)
```
Comment (jsturdy) [inline ultraLatency.py]: there is pretty much no case where when debugging a tool i'm interested in seeing the `uhal::LogLevel::DEBUG` messages, `uhal::LogLevel::INFO` should be sufficient

File: ultraLatency.py [hunk 4]
```diff
@@ -117,12 +114,10 @@
...
 N_EVENTS = Nev[0]
 
-mask = options.vfatmask
-
 try:
-    writeAllVFATs(ohboard, options.gtx, "ContReg0",    0x37, mask)
+    writeAllVFATs(ohboard, options.gtx, "ContReg0",    0x37, options.vfatmask)
     writeAllVFATs(ohboard, options.gtx, "ContReg2",    ((options.MSPL-1)<<4))
-    writeAllVFATs(ohboard, options.gtx, "VThreshold2", options.vt2, mask)
+    writeAllVFATs(ohboard, options.gtx, "VThreshold2", options.vt2, options.vfatmask)
```
Comment (jsturdy) [inline ultraLatency.py]: why this change? is it not clear what `mask` was?

Comment (bdorney) [reply ultraLatency.py]: See old line 120, a new variable `mask` was being used to store `options.vfatmask` which I felt was unnecessary.

File: fastLatency.py (hunk 1 above)
Comment (bdorney) [reply fastLatency.py]: kk, I'll change them all to `INFO`

File: ultraLatency.py (hunk 4 above)
Comment (jsturdy) [reply ultraLatency.py]: No, I saw what you did, I just wasn't clear on why you felt it was necessary (unnecessary).
* If you made separate commits (which would have been best) for the two changes you made (debug handling and `mask`), can you `cherry-pick` out the one related to the `mask`
* If not from a separate commit, then please revert the `mask` change
Then, you can create an issue if you feel it's worth changing the `mask` part

**Comment by bdorney** [comment]
@jsturdy requested revert has been performed

**Review by jsturdy — APPROVED** [review]
not sure all the added debug calls are desired...

**Comment by bdorney** [comment]
> not sure all the added debug calls are desired...

I guess my thought was better to have the option to have more information when debugging rather than the user having to make local changes to obtain that debugging information.

If it becomes a annoying or is ineffective we can open an issue to revert this.
